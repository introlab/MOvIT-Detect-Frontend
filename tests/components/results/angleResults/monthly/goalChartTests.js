/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import toJson from 'enzyme-to-json';
import GoalChart from '../../../../../src/components/results/angleResults/monthly/goalChart';

Enzyme.configure({ adapter: new Adapter() });

describe('GoalChart Tests', () => {
  const props = {
    condition: true,
    title: 'This is a test',
    successMessage: 'Success',
    data: {},
    options: {},
    isLoaded: true,
    hasErrors: false,
  };

  it('should have proptypes', () => {
    const actualValue = GoalChart.propTypes;

    const expectedValue = {
      condition: PropTypes.bool.isRequired,
      title: PropTypes.string.isRequired,
      successMessage: PropTypes.string.isRequired,
      data: PropTypes.object,
      options: PropTypes.object.isRequired,
      id: PropTypes.string,
      isLoaded: PropTypes.bool.isRequired,
      hasErrors: PropTypes.bool.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<GoalChart {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
