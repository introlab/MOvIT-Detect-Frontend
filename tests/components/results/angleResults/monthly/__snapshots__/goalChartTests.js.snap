// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalChart Tests should match the snapshot 1`] = `
<div>
  <div>
    <CustomCard
      element={
        <Chart
          className={null}
          data={Object {}}
          height={null}
          id={null}
          options={Object {}}
          style={null}
          type="line"
          width={null}
        />
      }
      header={
        <div>
          <h2
            id=""
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            This is a test
          </h2>
          <h4>
            Success
          </h4>
        </div>
      }
    />
  </div>
</div>
`;
